website/common/script/ops/buy/buyMarketGear.js

```diff
@@ -0,0 +1,78 @@
+import content from '../../content/index';
+import get from 'lodash/get';
+import pick from 'lodash/pick';
+import splitWhitespace from '../../libs/splitWhitespace';
+import {
+  BadRequest,
+  NotAuthorized,
+  NotFound,
+} from '../../libs/errors';
+import handleTwoHanded from '../../fns/handleTwoHanded';
+import ultimateGear from '../../fns/ultimateGear';
+
+import {removePinnedGearAddPossibleNewOnes} from '../pinnedGearUtils';
+
+import { AbstractGoldItemOperation } from './abstractBuyOperation';
+
+export class BuyMarketGearOperation extends AbstractGoldItemOperation {
+  constructor (user, req, analytics) {
+    super(user, req, analytics);
+  }
+
+  multiplePurchaseAllowed () {
+    return false;
+  }
+
+  extractAndValidateParams (user, req) {
+    let key = this.key = get(req, 'params.key');
+    if (!key) throw new BadRequest(this.i18n('missingKeyParam'));
+
+    let item = content.gear.flat[key];
+
+    if (!item) throw new NotFound(this.i18n('itemNotFound', {key}));
+
+    super.canUserPurchase(user, item);
```

Comment:
why super.canUserPurchase instead of this.canUserPurchase?

Reply:
I thought that would be better so that as a dev we could see its in a different class, but I can change it back to `this`